### PR TITLE
Events/tests

### PR DESCRIPTION
**Pull request by mehranhadidi**

### CODE REVIEW

**Comment by mehranhadidi** [comment]
Please add TODO if you feel something missing

**Comment by mehranhadidi** [comment]
@shov I don't think **await** & **then** makes sense in the event dispatching. They are very useful for listeners. which can be implemented in the listener class itself. 

Imagine that we have a user registered email. Listeners are "send welcome email", "create a user profile", "add registration loyalty point" & ... What is the point of dispatch in this case? 
in each one it make sense but at the event-level don't think so.

We should discuss it.

**Comment by mehranhadidi** [comment]
- Now dispatcher can handle async, promises & non-promises listeners
- Tests added
- Sample listeners added

File: lib/App.js [hunk 1]
```diff
@@ -176,16 +176,15 @@ class App {
    * Start servers, jobs, run boot() on the service providers
    * @return {App}
    */
-  start() {
+  async start() {
```
Comment (shov) [inline lib/App.js]: I'm not sure we should make the start method async. TBD

**Pull request merged**